new_course/1_programming_fundementals/debugging.md

```diff
@@ -125,9 +125,31 @@ end
 double_words_in_phrase("This is a test")
 ~~~
 
-When your code executes and gets to `binding.pry`, it will open an IRB-like session in your terminal. You can then use that session to check the values of anything within the scope of where you included `binding.pry`.
+When your code executes and gets to `binding.pry`, it will open an IRB-like session in your terminal. You can then use that session to check the values of anything within the scope of where you included `binding.pry`. However, keep in mind that any code written *after* the `binding.pry` statement will not have been evaluated during the Pry session. In other words, when we add a `binding.pry` statement, we are effectively creating a breakpoint or a pause in our code.
 
-Using Pry for debugging achieves the same outcome as `puts` debugging: it allows you to confirm the assumptions you have about particular parts of your code. If your code is complex, Pry will probably allow you to debug quicker thanks to its interactive runtime environment. In such scenarios, Pry will be easier to interact with than having to add `puts` statements everywhere and re-running your code each time.
+To see this point in action, try running the following:
+
+~~~ruby
+# script.rb
```

Comment:
This bit seems confusing. All of this code will work fine in IRB, so could you delete this bit for now? If you want to give the option of doing this either in IRB or with an actual file, could you add those steps explicitly?

Reply:
I wrote that to keep it consistent with the example above it. But I'll go ahead and remove it from both examples.

Reply:
I didn't realize that, sorry! I went in and added some additional text about running the code either in IRB or from a file. Thanks for bringing that to my attention!